### PR TITLE
test: automate evm e2e tests

### PR DESCRIPTION
**Pull request by technicallyty**
## Overview

this PR automates the EVM e2e tests ~~and adds them to the CI.~~ <-- preps it for the CI. will fix once polaris issues resolved.

if you want to try on your local machine: `make e2e-evm`

### CODE REVIEW

**Comment by technicallyty** [comment]
drafting this, something is gone awry with my go configuration

File: e2e/tests/evm/evm_test.go [hunk 1]
```diff
@@ -15,7 +17,7 @@ import (
 )
 
 const (
-	adminNakamaID = "00000000-0000-0000-0000-000000000000"
+	adminNakamaID = "00000000-0110-0000-0000-000000000000"
```
Comment (jerargus) [inline e2e/tests/evm/evm_test.go:20]: What's the reason for this change?

**Review by jerargus — APPROVED** [review]
the `TestGetMessageByName` had a flaky test failure. It looks like the "getOpenPort" helper sometimes returns the same port. This update to NewTestFixture should make it more resilient:
```
	cardinalPort := getOpenPort(t)
	evmPort := cardinalPort
	for retries := 10; retries > 0; retries-- {
		evmPort = getOpenPort(t)
		if evmPort != cardinalPort {
			break
		}
		time.Sleep(10 * time.Millisecond)
	}
	assert.Assert(t, evmPort != cardinalPort, "failed to find different port for evm")
```

**Comment by technicallyty** [comment]
> cardinalPort := getOpenPort(t)
> 	evmPort := cardinalPort
> 	for retries := 10; retries > 0; retries-- {
> 		evmPort = getOpenPort(t)
> 		if evmPort != cardinalPort {
> 			break
> 		}
> 		time.Sleep(10 * time.Millisecond)
> 	}
> 	assert.Assert(t, evmPort != cardinalPort, "failed to find different port for evm")

with go1.22 you can make the `for` declaration much more concise:

```go
	for range 10 {
		evmPort = getOpenPort(t)
		if evmPort != cardinalPort {
			break
		}
		time.Sleep(10 * time.Millisecond)
	}
```
